pybricks/tools/pb_module_tools.c

```diff
@@ -226,31 +229,15 @@ static mp_obj_t pb_module_tools_run_task(size_t n_args, const mp_obj_t *pos_args
         mp_raise_msg(&mp_type_RuntimeError, MP_ERROR_TEXT("Run loop already active."));
     }
 
-    run_loop_is_active = true;
-
-    uint32_t start_time = mp_hal_ticks_ms();
-    uint32_t loop_time = pb_obj_get_positive_int(loop_time_in);
-
     mp_obj_iter_buf_t iter_buf;
     mp_obj_t iterable = mp_getiter(task_in, &iter_buf);
 
     nlr_buf_t nlr;
     if (nlr_push(&nlr) == 0) {
-
+        run_loop_is_active = true;
```

Comment:
Need to check if we should be doing this here or before calling `mp_getiter` above.

The thinking to move it here was to ensure `run_loop_is_active` is only set if `mp_getiter` doesn't raise, but we'll need to make sure the flag isn't set too late.

Reply:
Could potentially be a problem if someone did something like `run_task(wait(0))`.

Seems like it should be fine to move the `mp_get_iter()` inside of the nlr block too (after setting run loop active flag) to avoid this.

Reply:
I'll do that, but running `run_task(wait(123))` isn't going to work either way and didn't previously, since the argument is always evaluated before `run_task` starts.

 (This is a known limitation, but it isn't restrictive. Probably because the one use case where it does make sense, `run_task(multitask(main1(), main2()))`, does work).